### DISCUSSION TITLE
Getting gradients?

### DISCUSSION THREAD

**Discussion by arjun-mani** [InteractiveComputerGraphics/SPlisHSPlasH#198] [General]
Hi,

I was curious about the possibility of getting gradients from Splishsplash, i.e. if the simulator is completely differentiable. I know that's probably not the case right now. I think it could be very interesting from the perspective of control, design, and integration with deep neural networks. One example task is how to set my initial velocity so my fluid particles could end up in a particular location at time-step T. I am just wondering if this is something you might be interested in. How hard you think it might be, and whether you know any other frameworks in the meanwhile that support this capability. Thanks very much.

Sincerely,
Arjun

**Comment by janbender** [comment]
Hi, 

Currently, the simulator is not differentiable. In fact that is something we are interested to investigate in future. However, at the moment I cannot say how complex this is.
